hal/common/retarget.cpp

```diff
@@ -98,6 +98,9 @@ static void init_serial() {
 #if DEVICE_SERIAL
     if (stdio_uart_inited) return;
     serial_init(&stdio_uart, STDIO_UART_TX, STDIO_UART_RX);
+#if MBED_CONF_CORE_STDIO_BAUD_RATE
```

Comment:
Nope you're right. The goal here was a change as small as possible to move forward on cleaning up example programs.

I was planning on submitting another pr to insure 9600 is the baseline, including going through [these](https://github.com/mbedmicro/mbed/search?utf8=%E2%9C%93&q=serial_baud%28obj%2C+9600&type=Code) as you mentioned earlier.